edx_rest_framework_extensions/permissions.py

```diff
@@ -1,9 +1,71 @@
 """ Permission classes. """
+from oauth2_provider.ext.rest_framework.permissions import TokenHasScope
+from opaque_keys.edx.keys import CourseKey
+from rest_framework.exceptions import PermissionDenied
 from rest_framework.permissions import BasePermission
 
+from edx_rest_framework_extensions.decorators import (
+    jwt_authentication_required,
+    skip_unless_jwt_scopes_enforced,
+)
+from edx_rest_framework_extensions.utils import jwt_decode_handler
+
 
 class IsSuperuser(BasePermission):
     """ Allows access only to superusers. """
 
     def has_permission(self, request, view):
         return request.user and request.user.is_superuser
+
+
+class JwtHasScope(TokenHasScope):
```

Comment:
Where would you be checking the version number on the JWT Token?  For forward compatibility, we want to:

1. Declare somewhere (in a class variable perhaps) that JWT_TOKEN_READ_VERSION = 1
2. Reject tokens whose major version number > JWT_TOKEN_READ_VERSION

This allows us to modify JWT token formats or semantics in the future while knowing that clients will reject tokens that they can't understand.

Reply:
I'm not understanding how that check relates to permissions or scopes. Wouldn't something like that be more appropriate in `JwtAuthentication`?

Reply:
Why do we need a version number for this? Wouldn't the parsing code just raise an exception if it isn't able to parse a particular value from the token?